docs/workflow/argocd.md

```diff
@@ -0,0 +1,228 @@
+Github, but you can choose whatever platform with which you are most
+comfortable. For reference, we will be using [this repository] throughout the
+following steps. Because it is a public repository, you are welcome to also use
+it, but be aware that if you enable automatic deployment in the next steps on
+pushes to `MASTER`, new resources may be deployed into your environment.
+
+### Create an ArgoCD Project for Infrastructure
+
+The two primary organizational units in ArgoCD are [projects] and
+[applications]. For simplicity, a project can just be thought of a grouping of
+applications that adhere to the same policies. In this guide, we are going to
+create an `infra` project and and `apps` project because we want infrastructure
+administrators to have very different permission levels than those of
+application owners.
+
+To create a new project, go to `Settings > Projects > New Project` in the ArgoCD
+admin console. You may configure your project however you see fit, but it may be
+easiest to give the most permissive access for this example and narrow the scope
+if you intend to run in a long-term production scenario. Importantly, you must
+at least whitelist all Crossplane providers, managed resources, and resource
+classes that you intend to use as they are all cluster-scoped. In addition, you
+must at least enable `in-cluster` as a destination. An example of the most
+permissive project would look as follows: 
+
+![alt argo-infra-proj](argo-infra-proj.png)
+
+### Configure Dev and Prod Environments
+
+Before we create ArgoCD projects for our infrastructure environments, we need to
+consider how we are going to manage our `Provider` secrets in our cluster.
+ArgoCD offers a variety of [secrets management integrations], but we will create
+our secrets directly in our cluster for this guide. To set up your secrets and
+your `Provider` objects, follow the steps in the [GCP provider], [AWS provider],
+and [Azure provider] configuration guides. If you are not planning on deploying
+resources on any of the three platforms, you are welcome to skip the setup of a
+`Provider` for its stack.
+
+Now we need to set up `dev` and `prod` environment resources for GCP in our
+`infra` project. To do so, go to `Applications > New Application` and configure
+as shown below or as appropriate with your repository:
+
+![alt argo-infra-app-1](argo-infra-dev-app-1.png)
+
+![alt argo-infra-app-2](argo-infra-dev-app-2.png)
+
+After creating the application, you should see healthy status and a
+visualization of all resources that were created:
+
+![alt argo-infra-app-viz](argo-infra-dev-app-viz.png)
+
+You will notice that resource classes have both a `dev` and `prod` version
+created, while the network resources (i.e. `Network`, `Subnetwork`,
+`GlobalAddress`, `Connection`) have only one version and it is a managed
+resource. These resources do not support dynamic provisioning and do not have
```

Comment:
would expect separate dev and prod networks/subnets ... this is probably fine for the example, but would be a stronger example with separate IMO

Reply:
I definitely prefer the strongest example :) However, the one drawback I see to this is that we are creating the network resources statically so we will provision a prod network in addition whether a use wants to use this guide to deploy prod and dev or not. What would you think about just adding a sentence here to the effect of "In a production scenario, you may want to set up separate networking resources for prod and dev."?

Reply:
I have added a similar sentence in the most recent revision

Reply:
sounds good.